Parser/tokenizer.c

```diff
@@ -2091,12 +2164,7 @@ _PyTokenizer_FindEncodingFilename(int fd, PyObject *filename)
     const char *p_end = NULL;
     char *encoding = NULL;
 
-    fd = _Py_dup(fd);
```

Comment:
Hummmmm. This is adding a ton of new code to remove the call to `_Py_dup`. What would be the advantage here?